src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -492,15 +518,65 @@ private async Task ExecuteClusterTransactionTask()
             }
         }
 
+        struct ClusterTransactionBatch : IDisposable
+        {
+            private readonly ClusterTransactionCommand.SingleClusterDatabaseCommand[] _data;
+            public int Count = 0;
+
+            public ClusterTransactionBatch(int maxSize)
+            {
+                _data = ArrayPool<ClusterTransactionCommand.SingleClusterDatabaseCommand>.Shared.Rent(maxSize);
+            }
+
+            public ArraySegment<ClusterTransactionCommand.SingleClusterDatabaseCommand> GetData()
+                => new ArraySegment<ClusterTransactionCommand.SingleClusterDatabaseCommand>(_data, 0, Count);
+
+            public void Add(IEnumerable<ClusterTransactionCommand.SingleClusterDatabaseCommand> data)
+            {
+                foreach (var command in data)
+                {
+                    Add(command);
+                }
+            }
+            
+            public void Add(ClusterTransactionCommand.SingleClusterDatabaseCommand command) => _data[Count++] = command;
```

Comment:
do you need to check boundaries here?

Reply:
if you try to add more than `maxSize` ?

Reply:
I think it is unnecessary to check here while it was implemented for this specific task and not for general purposes.
The max is 256 and we read up to 256.
But if you think it is required I will add

Reply:
let's add it.
better safe than sorry here

Reply:
Done